CONTRIBUTING.rst

```diff
@@ -147,8 +147,7 @@ If you plan on doing development and testing, you will also need to install the
 
 .. code-block:: bash
 
-    pip install -r doc-requirements.txt
-    pip install -r test-requirements.txt
+    pip install -r requirements/test-requirements.txt
```

Comment:
Actually, I find the ordering here a bit annoying. When I read this doc, I follow the order to install `ip install -e .[extras]` first. After that, I read `If you plan on doing development and testing, you will also need to install the following into the conda environment`, where it asks you to install `requirements/test-requirements.txt` first and then `extras`. The result is, many libs are installed, uninstalled, and reinstalled back and forth (since some requirements have version limitations while others don't).

It would be convenient if you rephrase it to avoid developers installing `pip install -e .[extras]` first or make the requirements compatible (installing `test-requirements.txt` should not need to uninstall libs from `extras`).

Reply:
Good point. Can ew just merge these code two blocks into one?

```bash
conda create --name mlflow-dev-env python=3.6
conda activate mlflow-dev-env
# installs mlflow from current checkout with packages required for development
pip install -e .[extras] -r requirements/test-requirements.txt
pip install -e tests/resources/mlflow-test-plugin
```

Reply:
I'm not sure whether this will resolve the difficulty, since the audience might still install `[extras]` first and then have to uninstall the dependencies that do not meet `requirements/test-requirements.txt`.
Anyways, it just takes a longer time but is correct, so we don't have to spend too much time on it.